frontend/src/metabase/query_builder/components/view/QuestionRowCount.jsx

```diff
@@ -70,11 +101,11 @@ const QuestionRowCount = ({
           <LimitPopover
             className="p2"
             limit={limit}
-            onChangeLimit={limit => {
-              if (limit > 0) {
-                onQueryChange(query.updateLimit(limit));
+            onChangeLimit={nextLimit => {
+              if (nextLimit > 0) {
+                onChangeLimit(nextLimit);
               } else {
-                onQueryChange(query.clearLimit());
+                onChangeLimit(null);
               }
```

Comment:
This is one of my pet peeves 😜  how about
```ts
onChangeLimit{nextLimit => {
  onChangeLimit(nextLimit > 0 ? nextLimit : null);
}}
```
?

Reply:
Makes total sense, fixed!